### DISCUSSION TITLE
Decoupling text height from text box height so we can set textbox height independently

### DISCUSSION THREAD

**Discussion by soham-bhosale** [fabricjs/fabric.js#9913] [Q&A] [answered]
What i basically want to achieve is, instead of scaling textbox vertically, which stretches the text, increase the box height and keep the font size as before.

By doing this, I would get a textbox which has more height then that of text.
From what I see, the height of textbox is always linked to that of text.

Following is the fiddle i came up with for this.
https://jsfiddle.net/dextroz0808/xmh5zkrd/57/
It has some issues tho, somehow font is displayed outside the textbox once you scale long enough.

Check out this video ->

https://github.com/fabricjs/fabric.js/assets/120098376/8a1827ee-1bf8-4709-9cba-ccad3a0698b7

I think there might be more things where textbox height will be reset back to that of text.

By giving independent height to textbox, I can then introduce feature like vertical alignment, max lines, max characters and min font size etc.

But for all that to work, I need a solid way to set up height.

Please help me out here.

**Comment by mrienstra** [comment]
Now that v6 is out of beta, you may want to mention that you're using [v5.3.0](https://github.com/fabricjs/fabric.js/releases/tag/v5.3.0) (released Dec 20, 2022). Might as well use the latest v5 release, [v5.3.1](https://github.com/fabricjs/fabric.js/releases/tag/v5.3.1) (Nov 29, 2022).

If migrating to v6 is an option, it seems like all of the development energy is focused there, so you may get more support as well.

**Reply by soham-bhosale** [reply]
Yeah, you are right. Might as well upgrade now as we are not that deep into this.

**Reply by mrienstra** [reply]
The v6 docs haven't been released yet, but you can run them locally, better than nothing: https://github.com/fabricjs/fabricjs.github.io

And of course there's this:
https://github.com/fabricjs/fabric.js/issues/8299

**Reply by mrienstra** [reply]
Oh, whoops, I guess you don't need to run the new v6 docs locally, they can be found at:
https://fabricjs.github.io/

**Reply by soham-bhosale** [reply]
Thanks. New doc site looks good. :fire:

**Reply by asturur** [reply]
the website will still be fabricjs.com as soon as i finish adding stuff on fabricjs.github.io, i m trying to rationalize what is there what is useful and what no. Is a slow process.

**Reply by soham-bhosale** [reply]
Cool. Let us know if we can help you in any way. Even if its just small tasks or testing that you need to get done.

**Comment by mhieupham1** [comment]
https://github.com/user-attachments/assets/bd3d93ca-7d88-4432-a88f-df518d5c295d

is it ?

**Reply by mhieupham1** [reply]
@soham-bhosale

**Reply by soham-bhosale** [reply]
Yes! Exactly as such.
I have to add few more layers of logic on top but this as base is required.
I am not sure tho how you are handling the wrapping and how it will behave when you have words with spaces.
Still as for box controls, its the exact thing that I was referring to.

**Reply by mhieupham1** [reply, marked as answer]
i override the 'br' controls of Textbox. See my example code : https://jsfiddle.net/mhieupham/zdmb86t2/12/

**Reply by mhieupham1** [reply]
@soham-bhosale

**Reply by soham-bhosale** [reply]
@mhieupham1 
Thnx! I will check it out.

**Reply by mauricekindermann** [reply]
Outstanding. Exactly what I was looking for. Now I will see if I can get vertical alignment working.
